hooks/pre/31-mongo_storage_engine.rb

```diff
@@ -0,0 +1,53 @@
+pulp_db_param = param('katello', 'pulp_db_name') || param('foreman_proxy_content', 'pulp_db_name')
+fail_and_exit('Unknown pulp database name') unless pulp_db_param
```

Comment:
This shouldn't be in the global scope. Below is an `if app_value(:upgrade_mongo_storage_engine)` and only then should it actually fail. Otherwise the code isn't even executed.

I'd move all 3 lines into the `migration` function.

Reply:
+1 will move.